src/repository/opamRepository.ml

```diff
@@ -396,10 +396,12 @@ let apply_repo_update repo = function
     Done ()
   | Update_err _ -> assert false
 
-let cleanup_repo_update = function
-  | Update_full d -> OpamFilename.rmdir d
-  | Update_patch f -> OpamFilename.remove f
-  | _ -> ()
+let cleanup_repo_update upd =
+  if not (OpamConsole.debug ()) then
+    match upd with
+    | Update_full d -> OpamFilename.rmdir d
+    | Update_patch f -> OpamFilename.remove f
+    | _ -> ()
```

Comment:
using `git apply` could actually be a good solution, if it's more reliable. At the moment we require `patch`, but not `git`, though.
Part of the plan behind the new update feature was to use a patch-file generator written in OCaml, common to Conex and opam. Since patches can be supplied in packages, though, we need to support the number of different patch formats out there, and it's wiser to rely on an external `patch` command for that.